### PR TITLE
Add user activity metrics

### PR DESCRIPTION
**Pull request by Half-Shot**
This should be equivalent to https://github.com/matrix-org/matrix-appservice-slack/pull/380 (sans rooms)

We're basically just cribbing off our postgres data and inserting it into prometheus

### CODE REVIEW

File: src/bridge/IrcBridge.ts [hunk 1]
```diff
@@ -296,6 +303,31 @@ export class IrcBridge {
...
+                        else if (appserviceBot.isRemoteUser(user.user_id)) {
+                            remote++;
+                        }
+                        else {
+                            matrix++;
+                        }
+                    }
+                    activeUsers.set({remote: "true"}, remote);
+                    activeUsers.set({remote: "false"}, matrix);
+                });
```
Comment (jaller94) [inline src/bridge/IrcBridge.ts:328]: Needs a `.catch()`, otherwise we may end up with an unhandled promise.

File: src/bridge/IrcBridge.ts [hunk 2]
```diff
@@ -685,9 +717,12 @@ export class IrcBridge {
     private async _onEvent (baseRequest: Request): Promise<BridgeRequestErr|undefined> {
         const event = baseRequest.getData();
         let updatePromise: Promise<void>|null = null;
-        if (event.sender && this.activityTracker) {
+        if (event.sender && (this.activityTracker ||
+            this.config.ircService.metrics.userActivityThresholdHours !== undefined)) {
```
Comment (jaller94) [inline src/bridge/IrcBridge.ts:721]: * `ircService` may not be defined according to the schema.
* `metrics` may not be defined according to the schema.

File: src/bridge/IrcBridge.ts [hunk 3]
```diff
@@ -266,6 +267,12 @@ export class IrcBridge {
             labels: ["server"]
         });
 
+        const activeUsers = metrics.addGauge({
+            name: METRIC_ACTIVE_USERS,
+            help: "Numer of users actively using the bridge.",
```
Comment (jaller94) [inline src/bridge/IrcBridge.ts:272]:
```suggestion
            help: "Number of users actively using the bridge.",
```

File: src/bridge/IrcBridge.ts (hunk 2 above)
Comment (Half-Shot) [reply src/bridge/IrcBridge.ts:721]: >ircService may not be defined according to the schema.

That's bad, it definitely should be defined.

> metrics may not be defined according to the schema.

That sound possible, our types are wrong.

**Pull request merged**